navigation/src/main/java/com/example/navigation/bottomAppNavigation/BottomAppNavigation.kt

```diff
@@ -23,47 +26,31 @@ import kotlinx.coroutines.flow.collectLatest
 @Composable
 fun BottomAppNavigation(
     ///Bottom navigation composable function that takes a
-    navigator: Navigator,
 
     ///Define the homeScreen, documentScreen, and searchScreen composable functions as parameters.
     homeScreen: @Composable () -> Unit,
     documentScreen: @Composable () -> Unit,
     searchScreen: @Composable () -> Unit
 ) {
-    val navController = rememberNavController()
 
-    ///Define a rememberDestination mutableState variable to store the current destination.
     val rememberDestination = remember { mutableStateOf(Destination.home.route) }
 
-    LaunchedEffect(Unit) {
-        navigator.actions.collectLatest {
-            when (it) {
-                is Navigator.NavigationActions.Navigate -> {
-                    navController.navigate(it.destination, builder = it.navOptions)
-                }
-
-                is Navigator.NavigationActions.Back -> {
-                    navController.popBackStack()
-                }
-            }
-        }
-    }
+    val state = rememberPagerState(initialPage = 0, pageCount = { bottomNavigationTabs.size })
 
     BottomNavigationWrapper(
         currentDestination = rememberDestination.value,
         body = {
-            NavHost(
-                navController = navController,
-                startDestination = Destination.home.route
-            ) {
-                composable(Destination.home.route) { homeScreen() }
-                composable(Destination.document.route) { documentScreen() }
-                composable(Destination.search.route) { searchScreen() }
+            HorizontalPager(state = state) { page ->
+                when (page) {
+                    0 -> homeScreen()
+                    1 -> documentScreen()
+                    2 -> searchScreen()
+                }
```

Comment:
The use of hardcoded indices in the `when` block can lead to maintenance issues if the order of tabs changes or more tabs are added.

```diff
-                when (page) {
-                    0 -> homeScreen()
-                    1 -> documentScreen()
-                    2 -> searchScreen()
+                when (bottomNavigationTabs[page].route) {
+                    Destination.home.route -> homeScreen()
+                    Destination.document.route -> documentScreen()
+                    Destination.search.route -> searchScreen()
```